vllm/model_executor/guided_decoding/reasoner/reasoner.py

```diff
@@ -0,0 +1,31 @@
+# SPDX-License-Identifier: Apache-2.0
```

Comment:
I feel like this abstraction is a bit too complex. Can we get away with just propagate the reasoning config to "get_local_guided_decoding_logits_processor" and "get_guided_decoding_logits_processor" and compute `<think>` and `</think>` there?

we can cache it for xgrammar and outlines then instead of having another abstraction reasoning separate for guided decoding.

Reply:
Do you mean removing the reasoner abstraction and get the token ids in the `get_local_guided_decoding_logits_processor` / `get_guided_decoding_logits_processor`?

Reply:
yeah, because we have access to the tokenizer there

Reply:
Yeah, that’s where I started with this. But honestly, I’m not sure if we’ll need to add more reasoning models down the line, and the end_think_token could change too. So I made this abstraction easy to extend later.

E.g. https://github.com/vllm-project/vllm/issues/13342

Reply:
Maybe we could merge the reasoner and the reasoner config into one abstraction. 

```python
def get_reasoning_config(tokenizer, reasoning_backend):
    if reasoning_backend == "deepseek":
        reasoner_config = ReasonerConfigDeepSeek.from_tokenizer(tokenizer)
    return reasoner_config

async def get_guided_decoding_logits_processor(
        guided_params: GuidedDecodingParams,
        tokenizer: PreTrainedTokenizer,
        model_config: ModelConfig,
        reasoning_backend: str | None = None) -> LogitsProcessor | None:
...
        reasoner_config = get_reasoning_config(tokenizer, reasoning_backend)
...
```

Reply:
I merged the reasoner and the reasoner config into one abstraction. It should be cleaner.

Reply:
yeah I think we can get the Config like this.

ig once s1 type of models are supported then we would run into different parsing. Oh well.